File: python/grpc_client.py
import grpc

from proto.hello_pb2 import HelloInput, HelloOutput
from proto.hello_pb2_grpc import HelloStub


if __name__ == '__main__':
    with grpc.insecure_channel('localhost:9988') as channel:
        stub = HelloStub(channel)
        request = HelloInput(name='ytdu', repeat=3)
        response = stub.Hello(request, None)
        print(response.hello_message)

File: cpp/grpc_client.cc
#include <iostream>
#include <grpc++/grpc++.h>
#include "proto/hello.grpc.pb.h"
#include "proto/hello.pb.h"


int main() {
  auto channel = grpc::CreateChannel("localhost:9988", grpc::InsecureChannelCredentials());
  auto stub = hello::Hello::NewStub(channel);
  grpc::ClientContext context;
  auto request = hello::HelloInput();
  request.set_name("ytdu");
  request.set_repeat(3);
  auto result = hello::HelloOutput();
  stub->Hello(&context, request, &result);
  std::cout << result.hello_message() << std::endl;
  return 0;
}
